### ISSUE TITLE
Suggestion: Allow users to retain Email Log table when plugin is deleted

### ISSUE THREAD

**Issue by mariadanieldeepak** [sudar/email-log#89]
This way, user can re-install the plugin at a later point, and still have the data.

**Comment by sudar** [comment]
That's a good suggestion.

Do you want to explore it further and come up with an implementation plan?

**Comment by mariadanieldeepak** [comment]
Thanks 👍 and I'm on it.

**Comment by mariadanieldeepak** [comment]
**Here is the plan, after my analysis.**

- Create a Settings section to allow the user retain the table while plugin uninstallation.
- Capture the user's selection in `options` table.
- The `option` value is checked before deleting the `email_log` table in the following line https://github.com/sudar/email-log/blob/master/uninstall.php#L20
- The plugin already handles the table creation when it is re-installed on the same site from which it was deleted (https://github.com/sudar/email-log/blob/master/include/Core/DB/TableManager.php#L248)

**Optimization**

- The `option` value that was set earlier, could be deleted after the table deletion is handled in `uninstall.php`. This way, we don't leave the `option` to become an Orphan record.

Kindly let me know your thoughts.

**References**

- https://developer.wordpress.org/plugins/the-basics/uninstall-methods/
- http://sudarmuthu.com/blog/lessons-from-wordpress-plugin-competition/

**Comment by sudar** [comment]
@mariadanieldeepak 

The design is good. Let's implement it.

Also we already have a settings screen created for the auto delete logs add-on. If I remember correctly it is extendable. Let's use the same settings screen.

**Comment by sudar** [comment]
Something to consider.

Easy Digital Downloads has the following settings. We can do something similar.

<img width="942" alt="screenshot 2017-08-14 18 11 46" src="https://user-images.githubusercontent.com/103148/29272089-3d9d57e2-811c-11e7-93ab-81bece4c1e53.png">

**Comment by mariadanieldeepak** [comment]
@sudar,

This has been implemented and a PR has been sent - https://github.com/sudar/email-log/pull/102

**Testing:**

![image](https://user-images.githubusercontent.com/17510033/29906606-afbd3634-8e33-11e7-97af-283be4fac5f3.png)

By default when the plugin is deactivate and deleted (from WP Admin), the Email Log table is not deleted. When the "Remove Data on uninstall?" option is checked, the table is removed.

Additionally, I've included a link to an add-on and ensured that a potential security vulnerability is handled - https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

Kindly let me know your comments.
